core/src/main/java/org/openstack4j/openstack/storage/block/domain/CinderEncryptionType.java

```diff
@@ -0,0 +1,163 @@
+package org.openstack4j.openstack.storage.block.domain;
+
+import org.openstack4j.model.storage.block.EncryptionType;
+import org.openstack4j.model.storage.block.builder.EncryptionTypeBuilder;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+import com.google.common.base.MoreObjects;
+
+/**
+ * The encryption type defines the characteristics of a encryption.
+ *
+ * @author Jyothi Saroja
+ */
+@JsonRootName("encryption")
+public class CinderEncryptionType implements EncryptionType {
+
+    private static final long serialVersionUID = 1L;
+
+    @JsonProperty("encryption_id")
+    protected String id;
+    @JsonProperty("provider")
+    protected String provider;
+    @JsonProperty("control_location")
+    protected String controlLocation;
+    @JsonProperty("cipher")
+    protected String cipher;
+    @JsonProperty("key_size")
+    protected int keySize;
+    @JsonProperty("volume_type_id")
+    protected String volumeTypeId;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getProvider() {
+        return provider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getVolumeTypeId() {
+        return volumeTypeId;
+    }
+
+    /**
+    * {@inheritDoc}
+    */
+    @Override
+    public String getCipher() {
+        return cipher;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getKeySize() {
+        return keySize;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getControlLocation() {
+        return controlLocation;
+    }
+
+    @Override
+    public EncryptionTypeBuilder toBuilder() {
+        return new ConcreteEncryptionTypeBuilder(this);
+    }
+
+    /**
+     * @return the Encryption Type Builder
+     */
+    public static EncryptionTypeBuilder builder() {
+        return new ConcreteEncryptionTypeBuilder();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this).omitNullValues().add("id", id).add("volumeTypeId", volumeTypeId)
+                .add("provider", provider).add("cipher", cipher).add("keySize", keySize)
+                .add("controlLocation", controlLocation).toString();
+    }
+
+    public static class ConcreteEncryptionTypeBuilder implements EncryptionTypeBuilder {
+
+        private CinderEncryptionType cinderEncryptionType;
+
+        ConcreteEncryptionTypeBuilder() {
```

Comment:
Is there any particular reason to make these package protected? I am fine with both public or private here.

Reply:
Same as above, just followed the convention in CinderVolume class.

Reply:
I see. We can disregard that for now as the codebase requires significant cleanup anyway https://github.com/openstack4j/openstack4j/projects/1.